src/tests/multihost/ad/test_access_control.py

```diff
@@ -0,0 +1,409 @@
+            1.Set  access_provider to permit in sssd.conf
+            2.Log in as any AD-user
+        :expectedresults:
+            1. SSSD should restart correctly
+            2. Log in should be successful
+        """
+        tools = sssdTools(multihost.client[0])
+        (aduser, adgrp) = create_aduser_group
+        domain_name = tools.get_domain_section_name()
+        dom_section = f'domain/{domain_name}'
+        sssd_params = {'access_provider': 'permit'}
+        tools.sssd_conf(dom_section, sssd_params)
+        tools.clear_sssd_cache()
+        _, stdout = ssh_login(multihost, f'{aduser}@{domain_name}')
+        assert aduser in stdout, 'ADuser log in failed'
+
+    @staticmethod
+    def test_deny_all_users(multihost, create_aduser_group, backupsssdconf):
+        """
+        :title: Set access provider to deny all users
+        :id: 36f82023-3029-4fe4-82b1-7390f01bb5d6
+        :description: Set access_provider to deny This should allow log in of any valid AD-user
+        :steps:
+            1.Set  access_provider to deny in sssd.conf
+            2.Log in as any AD-user
+        :expectedresults:
+            1. SSSD should restart correctly
+            2. Log in should be denied
+        """
+        (aduser, _) = create_aduser_group
+        tools = sssdTools(multihost.client[0], multihost.ad[0])
+        domain_name = tools.get_domain_section_name()
+        multihost.client[0].run_command('realm deny --all', raiseonerr=False)
+        tools.clear_sssd_cache()
+        _, stdout = ssh_login(multihost, f'{aduser}@{domain_name}')
+        assert stdout, 'Access denied log is absent in /var/log/secure'
+
+    @staticmethod
+    def test_dont_fail_auth_with_allow_rules(multihost, create_aduser_group, backupsssdconf):
+        """
+        :title: Set access_provider to permit all users
+        :id: 4e746a3b-8f8e-4884-930f-2ae8ef72e626
+        :description: When simple_allow_groups has one valid and one invalid group as inputs then
+         sssd should ignore invalid group argument. SSSD should allow member-users from valid group
+         to log in
+        :steps:
+            1. Define simple_allow_group with one valid group and one invalid group
+            2. Log in from that valid-groups's member-user
+        :expectedresults:
+            1. SSSD should restart correctly
+            2. Log in should be allowed
+        """
+        (aduser, adgroup) = create_aduser_group
+        tools = sssdTools(multihost.client[0], multihost.ad[0])
+        domain_name = tools.get_domain_section_name()
+        dom_section = f'domain/{domain_name}'
+        sssd_params = {'access_provider': 'simple',
+                       'simple_allow_groups': f'{adgroup}@{domain_name}, non_grp@{domain_name}'}
+        tools.sssd_conf(dom_section, sssd_params, action='add')
+        tools.clear_sssd_cache()
+        _, stdout = ssh_login(multihost, f'{aduser}@{domain_name}')
+        assert aduser in stdout, 'ADuser logged in failed'
```

Comment:
I would suggest to add a check with a user which is not member of `{adgroup}@{domain_name}` to check is it is rejected as expected.